### PR TITLE
Navigation: Move scroll behaviour to body

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- makes the body the scrollable element behind the `bodyScrolling` feature toggle
  - various CSS changes (behind toggle) to `body`, `.grafana-app`, `.main-view`, `AppChrome` and `Page`
  - effectively deprecates `NativeScrollbar` since it does nothing now when `bodyScrolling` is enabled
  - adjusts `DashboardPage` to scroll the window position instead of the container position
- adds a `useChromeHeaderHeight` hook to grafana/runtime to expose the header height so that plugins can position things stickily just below the header
  - uses the hook in `DataTrail` (explore metrics) 
  - uses the hook in `ConnectData/Search` (connections page) to stickily position the search under the header bar in OSS
    - in cloud, this uses an app plugin so will need a change to the plugin as well
- changes `LiveConnectionWarning` to be `fixed` position so it sits at the bottom of the page correctly
  - this change isn't toggled since it will behave the same both with and without the toggle on.
  - also converts to emotion object style
- updates unit tests to pull in config where needed

TODO

- [x] connections sticky header
  - [x] TODO review other `position: sticky;` elements
- [x] scrollRef/scrollTop on dashboard
- [x] ~scrollRef/scrollTop on scenes dashboards~
  - this doesn't work in main
- [x] panel edit in scenes
- [ ] Fix some plugin pages not spanning full page height:
  - [x] ~Incident -> Tasks~
    - this doesn't work in main
  - [ ] Kubernetes pages
  - [ ] AWS -> Logs
  - [ ] Application -> Configuration
  - [ ] Frontend -> Errors
- [x] verify enterprise pages
- [x] verify reporting 
- [x] verify notification banner works

**Why do we need this feature?**

- better for a11y (arrow keys now work to scroll up/down immediately on page load)
- better for mobile view (url bar hides correctly on mobile)
- better for taking full page screenshots

**Who is this feature for?**

- everyone

**Which issue(s) does this PR fix?**:

For https://github.com/grafana/grafana/issues/89894
For https://github.com/grafana/grafana/issues/13437

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by joshhunt** [comment]
@ashharrison90 Through testing this, I notice that all pages have a tiny (~3px?) bit of overscroll. Easily noticable on `/admin`. 

It's problematic on pages like tall dashboards where it can prevent the dashboard itself from scrolling.

Windows, Firefox with overlay scrollbars.

**Comment by ashharrison90** [comment]
i'll boot my windows pc up later, but i can't see that on mac at all:

image

File: public/app/core/components/NativeScrollbar.tsx [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
 import { useEffect, useRef } from 'react';
 
+import { config } from '@grafana/runtime';
 import { CustomScrollbar, useStyles2 } from '@grafana/ui';
 
 type Props = Parameters<typeof CustomScrollbar>[0];
 
 // Shim to provide API-compatibility for Page's scroll-related props
+// when bodyScrolling is enabled, this is a no-op
+// TODO remove this shim completely when bodyScrolling is enabled
```
Comment (joshhunt) [inline public/app/core/components/NativeScrollbar.tsx:11]: Unfortunately, I think we should maintain compatibility here, and have these properties return a ref(?) to the body, and scroll the body.

Ideally, we shouldn't have any changes to the DashboardPage for the scroll restoration.

**Review by joshhunt — COMMENTED** [review]
Tested this and spotted that one issue i mentioned above. Also smoke tested it with the feature toggle _off_ and it was all fine.

I think this is okay to merge (given it's flagged), but I think we should maintain compatibility with the scroll props on Page.

File: public/app/core/components/NativeScrollbar.tsx (hunk 1 above)
Comment (ashharrison90) [reply public/app/core/components/NativeScrollbar.tsx:11]: this is gonna sound sassier than i'm intending, but... why? 🤔

the only place that uses it is `DashboardPage` and it's not exposed to plugins via `PluginPage`. feels like this is a good time to decouple it from the `Page` interface. unless we're thinking `Page` is the correct place for this logic to live 🤔

**Review by joshhunt — APPROVED** [review]
Nice. Excellent work.

**Pull request merged**